pkg/ipam/types.go

```diff
@@ -12,6 +12,10 @@ import (
 	"github.com/davecgh/go-spew/spew"
 )
 
+const (
+	customPodIpAddr = "cni.cilium.io/ipAddrs"
```

Comment:
I don't have background on this proposal at all, but I noticed this so I figure I'll mention it: a minor nit, to ensure consistency between different annotations and make it easy to locate all annotations, we manage these through `pkg/annotation/k8s.go`. The format is also a bit different, the prefix should be `io.cilium`.

Reply:
Okay, how do you think this constant should be named

Reply:
use  `io.cilium.cni/ipAddrs` ?

Reply:
Sorry, I jumped ahead a bit and I think that we should explore @twpayne's questions before we come to a conclusion on this. The naming will depend on the semantics you are expecting from the feature but I don't yet have my head wrapped around the overall problem or how this PR fits into the solution.